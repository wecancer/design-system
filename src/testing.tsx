import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
