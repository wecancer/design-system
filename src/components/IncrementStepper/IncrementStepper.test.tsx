import IncrementStepper from '.'
import {render, fireEvent} from '../../testing'

describe('<IncrementStepper />', () => {
  it('should render snapshot correctly', () => {
    const {container} = render(
      <>
        <IncrementStepper value={10} onChange={() => null} />
      </>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should increment the value when the button is clicked', () => {
    const {container} = render(<IncrementStepper value={10} onChange={({value}) => expect(value).toBe(11)} />)
    const button = container.querySelector('button:last-child')

    if (!button) throw new Error('Last button not found')

    fireEvent.click(button)
  })

  it('should decrement the value when the button is clicked', () => {
    const {container} = render(<IncrementStepper value={10} onChange={({value}) => expect(value).toBe(9)} />)
    const button = container.querySelector('button:first-child')

    if (!button) throw new Error('First button not found')

    fireEvent.click(button)
  })

  it('should disable the increment button when the max value is defined', () => {
    const {container} = render(<IncrementStepper value={10} max={10} onChange={() => null} />)
    const button = container.querySelector('button:last-child')

    if (!button) throw new Error('Last button not found')

    expect(button).toBeDisabled()
  })

  it('should disable the decrement button when the max value is defined', () => {
    const {container} = render(<IncrementStepper value={10} min={10} onChange={() => null} />)
    const button = container.querySelector('button:first-child')

    if (!button) throw new Error('First button not found')

    expect(button).toBeDisabled()
  })

  it('should disable the controls', () => {
    const {container} = render(<IncrementStepper isDisabled value={0} min={-10} max={30} onChange={() => null} />)
    const btnFirst = container.querySelector('button:first-child')
    const btnLast = container.querySelector('button:last-child')
    const input = container.querySelector('input')

    if (!btnFirst) throw new Error('First button not found')
    if (!btnLast) throw new Error('Last button not found')
    if (!input) throw new Error('Last button not found')

    expect(btnFirst).toBeDisabled()
    expect(btnLast).toBeDisabled()
    expect(input).toBeDisabled()
  })
})
