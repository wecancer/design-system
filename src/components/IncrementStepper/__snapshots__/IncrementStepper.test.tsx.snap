// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncrementStepper /> should render snapshot correctly 1`] = `
.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 svg {
  width: 0.625rem;
  height: 0.625rem;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0;
  outline: none;
  border: none;
  background: transparent;
}

.c0 {
  display: grid;
  grid-template-columns: 32px 40px 32px;
  -webkit-column-gap: 0.5rem;
  column-gap: 0.5rem;
}

.c4 {
  text-align: center;
  font-family: Poppins,sans-serif;
  font-size: 1rem;
  border: none;
  background-color: transparent;
}

.c2 {
  width: 32px;
  height: 32px;
  border-radius: 50%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Poppins,sans-serif;
  border: 1px solid #EFF0F6;
  background-color: #EFF0F6;
  -webkit-transition: all ease 250ms;
  transition: all ease 250ms;
}

.c2:not(:disabled):hover {
  background-color: #FCFCFC;
}

.c2:active {
  color: #0080EB;
  border-color: #0080EB;
}

.c2:focus {
  color: #0080EB;
  border-color: #0080EB;
  box-shadow: 0 0 1px 6px #CEE2FF;
}

.c2:disabled {
  opacity: 0.7;
  cursor: default;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1 c2"
      type="button"
    >
      <div
        class="c3"
        data-icon-name="minus"
      >
        <span />
      </div>
    </button>
    <input
      class="c4"
      type="text"
      value="10"
    />
    <button
      class="c1 c2"
      type="button"
    >
      <div
        class="c3"
        data-icon-name="plus"
      >
        <span />
      </div>
    </button>
  </div>
</div>
`;
